### DISCUSSION TITLE
Forward  port in client via strongwan tunnel

### DISCUSSION THREAD

**Discussion by diegargon** [strongswan/strongswan#1319] [Q&A] [answered]
Hello

I setup a vpn/ipsec tunnel with strongwan  in my linux client (haven't access to the other side)

Works fine.

In the other side they have a https server, I want give access to that http server located in  10.180.100.5  to other users on my
net without need use a vpn client.

That mean, i want forward in my client the port 443 to that http server.

I manage do it easy with a web proxy after lots of hours trying with iptables.

but I would be interested in doing it purely with iptables

`
2: ens32: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:0c:29:2a:45:95 brd ff:ff:ff:ff:ff:ff
    altname enp2s0
    inet 172.20.1.228/22 brd 172.20.3.255 scope global ens32
       valid_lft forever preferred_lft forever
    inet 172.18.16.2/32 scope global ens32
       valid_lft forever preferred_lft forever
    inet6 fe80::20c:29ff:fe2a:4595/64 scope link
       valid_lft forever preferred_lft forever
`
ip route show table 220
` 
10.0.0.0/8 via 172.20.0.1 dev ens32 proto static src 172.18.16.2
`

Firewall is off/accepting all

any help? example? or anything that can help me. Thanks

**Comment by tobiasbrunner** [comment, marked as answer]
You need two NATs, a DNAT mapping the destination address for traffic to port 443 from your IP to the remote HTTP server's IP (i.e. to `10.180.100.5`), and an SNAT that maps that traffic's source IP to the virtual IP you received from the VPN server (i.e. to `172.18.16.2`) so that it matches the IPsec policies.

Something like this:

    # iptables -t nat -A PREROUTING -p tcp --dport 443 -j DNAT --to-destination 10.180.100.5
    # iptables -t nat -A POSTROUTING -p tcp --dport 443 -d 10.180.100.5 -j SNAT --to-source 172.18.16.2

**Reply by diegargon** [reply]
that work fine, the problem that  i forgot mention (sorry) is the virtual ip is dynamic assigned by the other side, 
that not will not work after a reconnect/reboot.

**Reply by tobiasbrunner** [reply]
You can install these rules dynamically in a custom [updown script](https://docs.strongswan.org/docs/5.9/plugins/updown.html) or [vici client](https://docs.strongswan.org/docs/5.9/plugins/vici.html).

**Reply by diegargon** [reply]
yep!
I just discovered that and do it using $PLUTO_MY_SOURCEIP

thanks
